assets/scss/components/_menu.scss

```diff
@@ -1,35 +1,71 @@
-
 /* ==========================================================================
    #MENU
    ========================================================================== */
 
 .c-menu {
-    
-}
+    @include container();
+    display: none;
+
+    @include bp(medium) {
+        display: block;
+        position: relative;
+        z-index: $z-index-low;
+    }
+
+    &.js-has-active-sub-menu::before {
+        background-color: $color-white;
+        content: '';
 
-.c-menu > nav {
-    display: grid;
-    grid-template-columns: auto 1fr;
-    gap: rem-calc(10px);
+        display: block;
+        position: absolute;
+        top: 0;
+        left: 0;
+        height: 100%;
+        width: 100vw;
+        margin-left: 50%;
+        transform: translateX(-50%);
+        z-index: $z-index-base;
+    }
 }
 
-.c-menu > nav ul {
-    @include list-bare();
+.is-open {
+    .c-menu {
```

Comment:
It should be more like 
```
.c-menu {
  .is-open & { ... }
}
```
as you are defining styles for `menu` not `is-open` class.

But I'd suggest using modifier class `c-menu--open` instead of nesting or applying `is-open` directly on `c-menu` to prevent unintentional behavior if any of the parents will have the `is-open` class